env0/data_projects.go

```diff
@@ -0,0 +1,54 @@
+package env0
+
+import (
+	"context"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataProjects() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataProjectsRead,
+
+		Schema: map[string]*schema.Schema{
+			"projects": {
+				Type:        schema.TypeList,
+				Description: "list of projects",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
```

Comment:
I think we need to provide also those

```typescript
isArchived?: boolean;
hierarchy: string;
parentProjectId?: string;
```

The last two are for creating custom logic around sub-projects.
 For the first one (`isArchived`) we may want to filter by it cy default. Can we?
 and if possible add a parameter to `include_archived_projects`.

Reply:
I see that we don't have the ability to filter via the API.
So the filter of `include_archived_projects` should be locally in the provider.
WDYS?

Reply:
> Will it return archived projects?

Yes

> What should be the default of include_archived_projects true or false?

`false`. Meaning we should filter them locally by default.
Add them to the state only if `include_archived_projects` is `true`
This makes more sense because I don't see users use the `archived` ones so much.

Reply:
BTW
`hierarchy` looks like `uuid|uuid|....`.
and `parentProjectId` is just uuid.

Reply:
@razbensimon - I have added the additional code you requested.

https://github.com/env0/terraform-provider-env0/pull/787/commits/52f69fa6368c657f6f90eae34e08a8ee4ff0e7f0